projects/reservoir/fv3/era_process.py

```diff
@@ -0,0 +1,177 @@
+            print(day,' is regridded.')
+            coarsened = coarsen(var,day)
+            print(day,' is coarsened.')
+            masked = mask(coarsened)
+            print(day,' is masked.')
+            masked.to_netcdf(path + var + '/masked/'+var+'_'+day+'.nc') 
+            print(day,' is done.')
+            
+def setup_fregrid():
+    #os.system("rm -rf fregrid-example")
+    os.system("mkdir -p fregrid-example/source-data")
+    os.system("mkdir -p fregrid-example/source-grid-data")
+    os.system("gsutil -m cp gs://vcm-ml-intermediate/2023-06-19-fregrid-example-data/*.nc fregrid-example/source-data/")
+    os.system("gsutil -m cp gs://vcm-ml-raw/2020-11-12-gridspec-orography-and-mosaic-data/C96/*.nc fregrid-example/source-grid-data/")
+    os.system("sudo docker run \
+            -v $(pwd)/fregrid-example:/work \
+            us.gcr.io/vcm-ml/post_process_run:latest \
+            make_hgrid \
+            --grid_type regular_lonlat_grid \
+            --grid_name /work/era5_lonlat_grid \
+            --nxbnds 2 \
+            --nybnds 2 \
+            --xbnds -180,180 \
+            --ybnds -90,90 \
+            --nlon 720\
+            --nlat 362")
+        
+    os.system("sudo docker run \
+        -v $(pwd)/fregrid-example:/work \
+        us.gcr.io/vcm-ml/post_process_run:latest \
+        make_solo_mosaic \
+        --num_tiles 1\
+        --dir /work \
+        --tile_file era5_lonlat_grid.nc \
+        --mosaic_name /work/era5_lonlat_grid_mosaic")
+                
+def check_time_steps_complete(path, var):    
+    #check if data is complete (all time steps are there)
+    time_step_list = []
+    for f in glob.glob(path+ var +'/downloaded/*.nc'):
+        dataset = xr.open_dataset(f)
+        time_step_list.append(dataset['time'].data)
+    dataset.close()
+    arrs = list(np.concatenate(time_step_list))
+    arrs.sort()
+    for i,date in enumerate(arrs[1:]):
+        if not (arrs[i]+np.timedelta64(1,'D')==date):
+            print('missing day after:', arrs[i], i)
+    print('all time steps checked.')
+    
+def merge_into_file(path,var,i,day):
+    #create one file per weekday
+    day_sublist = []
+    for f in glob.glob(path+ var +'/downloaded/*.nc'):
+        dataset = xr.open_dataset(f)
+        for time in range(len(dataset.time)):
+            data_at_day = dataset.isel(time=slice(time,time+1))
+            if data_at_day.time.dt.weekday[0] == i:
+                day_sublist.append(data_at_day)
+
+        dataset.close()
+    merged_data = xr.concat(day_sublist,dim='time')
+    sorted_data = merged_data.sortby('time')
+    sorted_data.to_netcdf(path + var + '/merged/' + var + '_' + day +'.nc')
+    save_nc_int32_time(path + var + '/merged/'+var+'_'+day+'.nc', '/home/paulah/fregrid-example/'+var+'_'+day+'_i32_time.nc')
+    #os.system("cp "+path+var+"/merged/"+var+"_"+day+"_i32_time.nc /home/paulah/fregrid-example")
+    
+def save_nc_int32_time(infile, outfile):
```

Comment:
`assign_coords` didn't work I think because xarray is doing some auto"magic" translation of the underlying cf-compliant netCDF time dimension, so any datetime get's translated to `int64` upon write.  This broke fregrid, hence the manual manipulation.

Reply:
I think you should be able to override that behavior in xarray by specifying the `dtype` encoding of the time variable before writing it out to a file:
```python
ds.time.encoding["dtype"] = np.int32
```
This will not handle the potential calendar conversion, however.  [There is a nice method for this now in xarray](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.convert_calendar.html),  but I think it was added after the version of xarray we have pinned in fv3net.